apix-impl/src/main/java/eu/xenit/apix/alfresco/metadata/NodeService.java

```diff
@@ -421,6 +426,56 @@ public NodeAssociations getAssociations(eu.xenit.apix.data.NodeRef ref) {
         return new NodeAssociations(getChildAssociations(ref), getParentAssociations(ref), getTargetAssociations(ref));
     }
 
+    @Override
+    public List<eu.xenit.apix.data.NodeRef> getRecursiveParents(eu.xenit.apix.data.NodeRef ref, eu.xenit.apix.data.NodeRef rootRef) {
```

Comment:
Please call these methods `getAncestors` (etc)

Reply:
Is there ever a use case where `rootRef` is **not** company home? When you know your e.g. grand child node, and you know your grandparent node, don't you also know all the ancestors inbetween?

Reply:
In Alfred Desktop it is possible to change the root to a lower level. For example you only want users to be able to use sites (which is a child of company home). 
I can also imagine that for performance reasons you may not want to return all ancestors if you already know the ancestors up to a certain node and you don't want to retrieve them again.
It may also be used to check if a node is the child of another node. For example, I set the rootRef to the node reference of Shared and in the result the node reference of the last ancestor is not Shared then i know that a node is not a child of Shared.

Reply:
Good, sounds like it is a use case indeed.